velox/dwio/common/tests/TypeTests.cpp

```diff
@@ -29,6 +29,19 @@ using namespace facebook::velox::dwio::common::typeutils;
 using facebook::velox::type::fbhive::HiveTypeParser;
 using facebook::velox::type::fbhive::HiveTypeSerializer;
 
+void assertEqualTypeWithId(
+    std::shared_ptr<const TypeWithId>& orig,
+    std::shared_ptr<const TypeWithId>& copy) {
+  EXPECT_EQ(orig->size(), copy->size());
+  for (auto idx = 0; idx < orig->size(); idx++) {
+    auto origTypeChild = orig->childAt(idx);
+    auto cutTypeChild = copy->childAt(idx);
+    EXPECT_TRUE(origTypeChild->type()->kindEquals(cutTypeChild->type()));
+    EXPECT_NE(origTypeChild, cutTypeChild);
```

Comment:
Remove this line. Orig and copy could be the same object.

Reply:
This is exactly why we made this fix. If they are the same object, that would mean actual schema would be modified when the copy is modified.